.github/workflows/on-pullrequest.yml

```diff
@@ -88,10 +88,28 @@ jobs:
       - name: Build
         run: ./build.ps1 Build -Configuration Debug
 
-      - name: Run Unit Tests
+      - name: Install Coverlet Reference
         if: success()
+        run: |
+          dotnet tool install --global coverlet.console
+          dotnet tool install --global dotnet-reportgenerator-globaltool
+
+      - name: Run Unit Tests
         run: ./build.ps1 UnitTest -Configuration Debug
 
+      - name: Generate Coverage Report
+        if: always()
+        run: |
+          reportgenerator -reports:"coverage.cobertura.xml" -targetdir:"coveragereport" -reporttypes:Html
+          ls
```

Comment:
Suggestion: add line 103 as a command in `build.ps`. Maybe a new command "coverage" that calls the UnitTest function and then runs the report generator.

Reply:
Done